### PR TITLE
 Move static metadata to declarative setup.cfg

### PR DESCRIPTION
**Pull request by webknjaz**
##### SUMMARY

**Note: This is a reincarnation of #37830 with changes pulled in from #22330 to demo declarative config capabilities to @abadger in particular.**

There's no project description @ [PYPI](https://pypi.org/project/ansible/2.5.0/) since v2.3.2.0. The last one I've found with long description is v2.3.1.0.

This is a suggestion of how to fix it by rendering README.md into the long_description.

##### ISSUE TYPE

 - Feature Pull Request ?
 - Bugfix Pull Request
 - Docs Pull Request ?

##### COMPONENT NAME
python distribution meta / setup.py

##### ANSIBLE VERSION
2.3.2.0‒2.5.0+

##### ADDITIONAL INFORMATION
has description: https://pypi.org/project/ansible/2.3.1.0/
doesn't: https://pypi.org/project/ansible/2.5.0/

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    Topic :: System :: Installation/Setup
+    Topic :: System :: Systems Administration
+    Topic :: Utilities
+keywords =
+    ansible
+    automation
+    infrastructure
+
+[options]
+python_requires = >=2.6,!=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*
```
Comment (mattclay) [inline setup.cfg]: 3.4 is also not supported.

Comment (webknjaz) [reply setup.cfg]: yea, missed that while focusing on other stuff :)

File: setup.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+##! /usr/bin/env python
```
Comment (mattclay) [inline setup.py:1]: Was this supposed to be a valid shebang?

**Comment by abadger** [comment]
Thanks for doing a proof of concept!  Unfortunately, after discussing this on slack, I don't think we're going to want to go this route.  It turns setup.py from a script with mostly declarative syntax and just enough imperative code to do what we need into a much larger imperative code base with an unclear boundary between what's standard information and what we'd need to replace somehow.

I thought I might be okay with having two sources of truth (setup.py for legacy and setup.cfg for the eventual ten years down the road) but nitzmahone brought up that we lose the dynamic dependency stuff if we do that.  It might also mean that installing the package using the setup.cfg without setup.py would be broken (because it wouldn't handle symlinks correctly).

So I'm going to close this.  Sorry that you spent so much time on it.

File: setup.py (hunk 2 above)
Comment (webknjaz) [reply setup.py:1]: (typo)

**Pull request closed without merging**